applications/ehr_query_llm/lmm/rag/ehr/create_ehr_db.py

```diff
@@ -0,0 +1,254 @@
+# limitations under the License.
+
+# BE SURE TO HAVE FHIR APP RUNNING BEFORE STARTING THIS SCRIPT
+
+import logging
+import signal
+import sys
+import time
+from threading import Thread
+from typing import Optional
+
+import zmq
+from langchain.schema.document import Document
+from langchain_community.embeddings import HuggingFaceBgeEmbeddings
+from langchain_community.vectorstores import Chroma
+
+from operators.ehr_query_llm.fhir.ehr_query import FHIRQuery
+from operators.ehr_query_llm.fhir.ehr_response import FHIRQueryResponse
+from operators.ehr_query_llm.message_handling import MessageReceiver
+
+# FHIR global vars
+topic_request = "ehr-request"
+topic_response = "ehr-response"
+busy = False
+resources_to_retrieve = ["Observation", "FamilyMemberHistory", "Condition", "AllergyIntolerance"]
+
+# EHR Patient ID for John Doe
+mrn = "108915cc-b9f2-47db-ad4b-0143151f4f61"
+
+# LLM summary global vars
+ehr_data = None
+
+# Stores EHR requests initiated from this instance
+ehr_request_ids = []
+
+# Vector db global vars
+EHR_FINETUNED_MODEL = "/workspace/volumes/models/bge-large-ehr-finetune"
+CACHE_FOLDER = "/workspace/volumes/models"
+PERSISTENT_FOLDER = "/workspace/holohub/applications/ehr_query_llm/lmm/rag/ehr/db"
+
+
+def get_ehr_data(
+    allow_reqested_only: Optional[bool] = True,
+):
+    global ehr_data
+    global ehr_request_ids
+
+    receiver = MessageReceiver(topic_response, "tcp://localhost:5601")
+    while True:
+        response = receiver.receive_json(blocking=True)
+        try:
+            ehr_response = FHIRQueryResponse.from_json(response)
+            logging.info(f"Got a response for request, id: {ehr_response.request_id}")
+            if ehr_response.request_id.casefold() not in ehr_request_ids:
+                # For now, only support request/response scenario
+                # In the future, may need to support unsolicilated messages
+                logging.warning(
+                    f"Processing response for a untracked request, id {ehr_response.request_id}"
+                )
+                if allow_reqested_only:
+                    continue
+            else:
+                ehr_request_ids.remove(ehr_response.request_id.strip())
+                logging.info(f"Removed processed tracked request, id {ehr_response.request_id}")
+
+            if len(ehr_response.patient_resources) > 0:
+                for key in ehr_response.patient_resources:
+                    logging.info(f"Documents retrieved: {len(ehr_response.patient_resources[key])}")
+                    ehr_data = ehr_response.patient_resources
+                return
+        except Exception as ex:
+            logging.error(f"Error parsing FHIR response {ex}")
+        time.sleep(1)
+
+
+def send_request():
+    global ehr_request_ids
+    context = zmq.Context()
+    socket = context.socket(zmq.PUB)
+    socket.bind("tcp://*:5600")
+    # Give ZeroMQ time to bind TODO determine why this is needed
+    time.sleep(1)
```

Comment:
Leave TODO or address in this PR?

Reply:
Updated the comments in code, to be in the next commit     
   # Give ZeroMQ subscribers time to bind.
    # Subscribers may be waiting to connect but it takes a finite time to complete binding
    # to the pub socket. If it is published immediately, a message may be lost
    # as ZeroMQ does not guarantee message delivery if the sub is not fully connected.